astropy/table/pprint.py

```diff
@@ -497,17 +500,15 @@ def _pformat_col_iter(
         auto_format_func = get_auto_format_func(col, pssf)
         format_func = col.info._format_funcs.get(col_format, auto_format_func)
 
-        if len(col) > max_lines:
+        if n_rows > max_lines:
             if show_length is None:
                 show_length = True
             i0 = n_print2 - (1 if show_length else 0)
```

Comment:
Out of scope, but...

```python 
i0 = n_print2 - int(show_length)
```

Reply:
I don't think I should touch anything that not strictly in scope here, at the very least while I'm waiting for @taldcroft to decide the fate of this patch :-)